Xamarin.PropertyEditing.Mac/Controls/Custom/MaterialView.cs

```diff
@@ -32,26 +32,17 @@ private void Initialize ()
 
 		public override void OnViewModelChanged (MaterialDesignColorViewModel oldModel)
 		{
-			NeedsLayout = true;
-		}
+			base.OnViewModelChanged (oldModel);
 
-		public override void OnPropertyChanged (object sender, PropertyChangedEventArgs e)
-		{
-			switch (e.PropertyName) {
-			case nameof (ViewModel.ColorName):
-			case nameof (ViewModel.AccentColor):
-			case nameof (ViewModel.NormalColor):
-				NeedsLayout = true;
-				break;
-			}
+			CreateColourPallette ();
 		}
 
-		public override void Layout ()
+		private void CreateColourPallette ()
 		{
-			if (Layer?.Sublayers != null) {
-				foreach (var l in Layer.Sublayers) {
-					l.RemoveFromSuperLayer ();
-					l.Dispose ();
+			if (Subviews.Length > 0) {
```

Comment:
Capture subviews in a local, otherwise this marshals the `Subviews` array twice.

Reply:
Done.